src/api/index.js

```diff
@@ -118,7 +118,9 @@ export const v3 = {
   getJson: async (url, accessToken) => {
     const response = await v3.call(url, v3.parameters(accessToken));
 
-    return response.json();
+    return response.ok
+      ? response.json()
+      : response.json().then(err => Promise.reject(err));
```

Comment:
In fact, `response.json().then(err => Promise.reject(err))` is the same with `response.json()`. You handle the error in `then` and throw out identically. Please correct me if I misunderstood your thinking.

Reply:
@chinesedfan 
this because when the response status is 4xx it always handled by `.then()`, that's why i give return `Promise.reject'  to make it handled by `.catch()`. But if you think there better way to distinct beetween 200 OK and 4xx Error, let me know.

Another suggestion from me is to implement Axios and Redux-Saga to the app for better handling failures.

Reply:
@dhamanutd I see. Then you can modify `getRepository` to call `v3.call`, and check `response.status`. Please refer to [`checkReadMe`](https://github.com/gitpoint/git-point/blob/master/src/repository/repository.action.js#L159).